app/models/vendor.rb

```diff
@@ -0,0 +1,47 @@
+class Vendor < ActiveRecord::Base
+  belongs_to :market
+  has_many :sales
+  has_many :products
+
+  validates_presence_of :name, :number_of_employees, :market_id
+  validates_numericality_of :number_of_employees, :market_id
+
+  def recent_sales_first(how_many = sales.count)
+    sorted_sales = sales.sort_by { |sale| sale.purchase_time }
+    sorted_sales.reverse!
```

Comment:
`sort_by` and `reverse!` works here, but you might want to checkout [`order`](http://apidock.com/rails/ActiveRecord/QueryMethods/order). Making the database handle it is usually much faster than ActiveRecord.